action/src/utils.ts

```diff
@@ -167,4 +174,92 @@ export class Utils {
     public static isWindows() {
         return platform().startsWith('win');
     }
+    public static async validatePlatfornUrl(jfrogUrl: string): Promise<void> {
+        const jfrogUrlFailure: string = 'JF_URL must point on your full platform URL, for example: https://mycompany.jfrog.io/, make sure the platform is up and running and accessible.'
+        //verify that the provided JFrog URL is valid and responsive
+        const pingUrl: string = jfrogUrl!.replace(/\/$/, '') + '/artifactory/api/system/ping';
+        const httpClient: HttpClient = new HttpClient();
+        let response: HttpClientResponse
+        try {
+             response = await httpClient.get(pingUrl);
+        }catch (error: any) {
+            throw new Error(jfrogUrlFailure + ', Error returned is ' + error.message);
+        }
+        if (response.message.statusCode == 200) {
+            const body: string = await response.readBody();
+            if (body == 'OK') {
+                return ;
+            }else
+                throw new Error(jfrogUrlFailure);
+        } else {
+           throw new Error(jfrogUrlFailure);
+        }
+    }
+
+    public static async getJfrogOIDCCredentials(jfrogUrl: string): Promise<void> {
```

Comment:
The function's name doesn't really match what it does - It doesn't return OIDC credentials from JFrog (the return value is void).

This function retrieves a JWT from GitHub and forwards it to the JFrog platform to obtain a valid access token. This process occurs only if the user hasn't supplied the `oidc-provider-name` argument.
I'd call it `setupOIDCTokenIfNeeded`.